### PR TITLE
Rename annotations

### PR DESCRIPTION
**Pull request by miladcognite**
####  [Add "legacy" prefix to annotation api](https://github.com/cognitedata/cognite-sdk-python-experimental/commit/b0311d01b4abb0b5d5510b9520b64e127406f29e)

Renamed `client.annotations.*` to `client.legacy_annotations.*`
and renamed corresponding (data) class and filenames accordingly.

#### [Refactor: Remove "V2" postfix from Annotation api](https://github.com/cognitedata/cognite-sdk-python-experimental/commit/cc6129de66b3a61388ccdbcbe52ef6d6d5208d21)

Renamed `client.annotations_v2.*` to `client.annotations.*`
and renamed corresponding (data) class and filenames accordingly.

#### [Doc: Update change log and update annotation doc](https://github.com/cognitedata/cognite-sdk-python-experimental/commit/200c61b192d6a8604b0b09135836cd81fe0ddcb5)

* Remove doc for legacy annotations (soon to be deprecated)

* Remove V2 postfix for the new annotation api

## Checklist

- [x] Changelog entry added in `CHANGELOG` or not relevant for this change
- [x] Version updated in `pyproject.toml` or not relevant for this change

### CODE REVIEW

File: cognite/experimental/data_classes/annotations.py [hunk 1]
```diff
@@ -1,93 +1,168 @@
...
+        data (dict): The annotation information. The format of this object is decided by and validated against the 'annotation_type' attribute.
+        status (str): The status of the annotation, e.g. "suggested", "approved", "rejected".
+
+        annotated_resource_type (str): Type name of the CDF resource that is annotated, e.g. "file".
+        annotated_resource_id (int, optional): The internal ID of the annotated resource. Either this field or the external ID of the linked resource is required (not both).
+        annotated_resource_external_id (str, optional): The external ID of the annotated resource. Either this field or the internal ID of the linked resource is required (not both).
+
+        creating_app (str): The name of the app from which this annotation was created.
+        creating_app_version (str): The version of the app that created this annotation. Must be a valid semantic versioning (SemVer) string.
+        creating_user: (str, optional): A username, or email, or name. This is not checked nor enforced. If the value is None, it means the annotation was created by a service.
```
[the remark this answers is not in the document]
Comment (miladcognite) [reply cognite/experimental/data_classes/annotations.py:29]: As discussed, let's update the migration scripts accordingly.

**Review by torbjornopheim — APPROVED** [review]
Looks good, as far as I can see 👍

**Review by cyriaquebrousse — APPROVED** [review]
Good change that will clarify the situation for many!

**Comment by miladcognite** [comment]
Changes from https://github.com/cognitedata/cognite-sdk-python-experimental/pull/313 are now also included. 
FYI: @peetcremer1 & @greenbech 

Background: https://cognitedata.slack.com/archives/CG10VQPFX/p1653051830876409

**Pull request merged**